subsys/logging/log_core.c

```diff
@@ -215,6 +202,23 @@ static inline void msg_finalize(struct log_msg *msg,
 	}
 }
 
+static inline void msg_finalize(struct log_msg *msg,
+				struct log_msg_ids src_level)
+{
+	unsigned int key;
+
+	msg->hdr.ids = src_level;
+	msg->hdr.timestamp = timestamp_func();
+
+	key = irq_lock();
+
+	log_list_add_tail(&list, msg);
+
+	irq_unlock(key);
+
+	z_log_msg_post_finalize();
+}
+
```

Comment:
Not really seen much value in here, is the other function re-used anywhere ? Also, one function is using `z_log` namespace and the other not.

Reply:
See PR message: `This is in preparation for v2 where it will be reused.` regarding `z_` it is a private function. There is an issue to fix that elsewhere in the logger (it is also tuckled by v2 where private functions are prefixed).